test/deployment/deployment.go

```diff
@@ -0,0 +1,133 @@
+// Copyright 2024
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package deployment
+
+import (
+	_ "embed"
+	"encoding/json"
+	"fmt"
+	"os"
+	"os/exec"
+
+	"github.com/Mirantis/hmc/test/utils"
+	"github.com/a8m/envsubst"
+	"github.com/google/uuid"
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	"gopkg.in/yaml.v3"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+)
+
+type ProviderType string
+
+const (
+	ProviderCAPI  ProviderType = "cluster-api"
+	ProviderAWS   ProviderType = "infrastructure-aws"
+	ProviderAzure ProviderType = "infrastructure-azure"
+
+	providerLabel = "cluster.x-k8s.io/provider"
+)
+
+type Template string
+
+const (
+	TemplateAWSStandaloneCP Template = "aws-standalone-cp"
+	TemplateAWSHostedCP     Template = "aws-hosted-cp"
+)
+
+//go:embed resources/aws-standalone-cp.yaml.tpl
+var awsStandaloneCPDeploymentTemplateBytes []byte
+
+//go:embed resources/aws-hosted-cp.yaml.tpl
+var awsHostedCPDeploymentTemplateBytes []byte
+
+func GetProviderLabel(provider ProviderType) string {
+	return fmt.Sprintf("%s=%s", providerLabel, provider)
+}
+
+// GetUnstructuredDeployment returns an unstructured deployment object based on
+// the provider and template.
+func GetUnstructuredDeployment(provider ProviderType, templateName Template) *unstructured.Unstructured {
+	GinkgoHelper()
+
+	generatedName := uuid.New().String()[:8] + "-e2e-test"
+	_, _ = fmt.Fprintf(GinkgoWriter, "Generated cluster name: %q\n", generatedName)
+
+	switch provider {
+	case ProviderAWS:
+		// XXX: Maybe we should just use automatic AMI selection here.
+		amiID := getAWSAMI()
```

Comment:
AMI is getting selected automatically using image lookup see our docs [here](https://github.com/Mirantis/hmc/blob/main/docs/aws/cluster-parameters.md#image-lookup). 
So this should be removed, so we can test that lookup works as intended.